src/System.IO.Pipelines.Networking.Tls/OpenSslConnectionContext.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System.IO.Pipelines.Networking.Tls.Internal.OpenSsl;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines.Networking.Tls
+{
+    public class OpenSslConnectionContext : ISecureContext
```

Comment:
This should not be public.  The model should choose the right thing for the platform you're on.

Allowing the provider to be requested by the user may be an option (an application explicitly choosing to use OpenSSL instead of SecureTransport on a mac, for example), but that should be a secondary goal.